modules/fedimint-ln-client/src/lib.rs

```diff
@@ -1888,94 +1893,105 @@ impl LightningClientModule {
         }
     }
 
-    pub async fn wait_for_ln_payment(
+    /// Subscribes to either a internal or external lightning payment and
+    /// returns `LightningPaymentOutcome` that indicates if the payment was
+    /// successful or not.
+    pub async fn await_outgoing_payment(
         &self,
-        payment_type: PayType,
-        contract_id: ContractId,
-        return_on_funding: bool,
-    ) -> anyhow::Result<Option<serde_json::Value>> {
-        match payment_type {
-            PayType::Internal(operation_id) => {
-                let mut updates = self
-                    .subscribe_internal_pay(operation_id)
-                    .await?
-                    .into_stream();
-
-                while let Some(update) = updates.next().await {
-                    match update {
-                        InternalPayState::Preimage(preimage) => {
-                            return Ok(Some(
-                                serde_json::to_value(PayInvoiceResponse {
-                                    operation_id,
-                                    contract_id,
-                                    preimage: preimage.consensus_encode_to_hex(),
-                                })
-                                .unwrap(),
-                            ));
-                        }
-                        InternalPayState::RefundSuccess { out_points, error } => {
-                            let e = format!(
-                                "Internal payment failed. A refund was issued to {out_points:?} Error: {error}"
-                            );
-                            bail!("{e}");
-                        }
-                        InternalPayState::UnexpectedError(e) => {
-                            bail!("{e}");
-                        }
-                        InternalPayState::Funding if return_on_funding => return Ok(None),
-                        InternalPayState::Funding => {}
-                        InternalPayState::RefundError {
-                            error_message,
-                            error,
-                        } => bail!("RefundError: {error_message} {error}"),
-                        InternalPayState::FundingFailed { error } => {
-                            bail!("FundingFailed: {error}")
-                        }
+        operation_id: OperationId,
+    ) -> anyhow::Result<LightningPaymentOutcome> {
+        let operation = self.client_ctx.get_operation(operation_id).await?;
+        let variant = operation.meta::<LightningOperationMeta>().variant;
+        if !matches!(
+            variant,
+            LightningOperationMetaVariant::Pay(LightningOperationMetaPay { .. })
+        ) {
+            bail!("Operation is not a lightning payment")
+        }
+
+        let mut final_state = None;
+
+        // First check if the outgoing payment is an internal payment
+        if let Ok(updates) = self.subscribe_internal_pay(operation_id).await {
```

Comment:
Instead of guessing if it's internal or external it appears we could read `variant.is_internal_payment` here? Feels like the `PayType` argument in the old fn was also redundant.